### PR TITLE
address comments, add nb observations

### PR DESCRIPTION
**Pull request by richardblythman**
Addresses review comments from #16. 
Ran black formatter to address some of the white space comments.
Removed cost-evaluations.ipynb, as it was a duplicate
Adds observations to the analysis notebooks, including costing.

### CODE REVIEW

File: fpmms/tools.py [hunk 1]
```diff
@@ -606,14 +633,22 @@ def pipeline_step(
         with ThreadPoolExecutor(max_workers=NUM_WORKERS) as executor:
             futures = []
             for i in range(0, len(parsed), 1000):
-                futures.append(executor.submit(get_contents, session, parsed[i:i+1000], event_name))
+                futures.append(
+                    executor.submit(
+                        get_contents, session, parsed[i : i + 1000], event_name
```
Comment (Adamantios) [inline fpmms/tools.py:638]: Maybe rename the batch size constant and use it here too?

File: fpmms/costing/costing.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
+import tiktoken
+import anthropic
+from typing import Optional
+from .prompts import prediction_request
+from .prompts import prediction_request_sme
+from .prompts import prediction_request_claude
```
Comment (Adamantios) [inline fpmms/costing/costing.py:6]: We keep a local hardcoded copy. Not ideal if the prompts change in the future. On the other hand, these tools are pretty much finalized so it should not be an issue.

What do you think @DavidMinarsch?

**Review by Adamantios — COMMENTED** [review]
LGTM, notebooks' comments help. Could you also add a short description on top of each notebook?

Also, let me know what you think re hardcoding the prompts.

File: fpmms/costing/costing.py (hunk 2 above)
Comment (DavidMinarsch) [reply fpmms/costing/costing.py:6]: I'd prefer that we imported the https://github.com/valory-xyz/mech repo as a subrepo and then we avoid the tool code duplication. @richardblythman can you do that?

File: fpmms/tools.py (hunk 1 above)
Comment (moarshy) [reply fpmms/tools.py:638]: sure

File: fpmms/costing/costing.py (hunk 2 above)
Comment (moarshy) [reply fpmms/costing/costing.py:6]: I think subrepo is a better option. Will work on this

Comment (richardblythman) [reply fpmms/costing/costing.py:6]: We did it this way before (with a symlink). Do you want me to change it back, and add some details to the readme for creating the symlink? Or using a git submodule? @DavidMinarsch @Adamantios

Comment (DavidMinarsch) [reply fpmms/costing/costing.py]: submodule please

**Comment by moarshy** [comment]
@Adamantios could you please give this a final look?

File: fpmms/tools.py [hunk 3]
```diff
@@ -647,29 +682,29 @@ def store_progress(
...
+                content.drop(columns=["error"], inplace=True)
 
-            if 'result' in content.columns:
-                content.drop(columns=['result'], inplace=True)
+            if "result" in content.columns:
+                content.drop(columns=["result"], inplace=True)
 
             content.to_csv(event_filename, index=False, escapechar="\\")
 
         # drop result and error columns
```
Comment (Adamantios) [inline fpmms/tools.py:693]:
```suggestion
        # drop result column
```

**Pull request merged**